### PR TITLE
Add SuppressWarnings SS_SHOULD_BE_STATIC

### PR DESCRIPTION
**Pull request by Fokko**
Fixes #XXXX.

(Replace XXXX with the id of the issue fixed in this PR. Remove the above line if there is no corresponding
issue. Don't reference the issue in the title of this pull-request.)

(If you are a committer, follow the PR action item checklist for committers:
https://github.com/apache/incubator-druid/blob/master/dev/committer-instructions.md#pr-and-issue-action-item-checklist-for-committers.)

### Description

Describe the goal of this PR, what problem are you fixing. If there is a corresponding issue (referenced above), it's
not necessary to repeat the description here, however, you may choose to keep one summary sentence.

Describe your patch: what did you change in code? How did you fix the problem?

If there are several relatively logically separate changes in this PR, create a mini-section for each of them. For
example:
#### Added a forbidden-apis entry ...

In each section, please describe design decisions made, including:
 - Choice of algorithms
 - Behavioral aspects. What configuration values are acceptable? How are corner cases and error conditions handled, such
   as when there are insufficient resources?
 - Class organization and design (how the logic is split between classes, inheritance, composition, design patterns)
 - Method organization and design (how the logic is split between methods, parameters and return types)
 - Naming (class, method, API, configuration, HTTP endpoint, names of emitted metrics)

It's good to describe an alternative design (or mention an alternative name) for every design (or naming) decision point
and compare the alternatives with the designs that you've implemented (or the names you've chosen) to highlight the
advantages of the chosen designs and names.

If there was a discussion of the design of the feature implemented in this PR elsewhere (e. g. a "Proposal" issue, any
other issue, or a thread in the development mailing list), link to that discussion from this PR description and explain
what have changed in your final design compared to your original proposal or the consensus version in the end of the
discussion. If something hasn't changed since the original discussion, you can omit a detailed discussion of those
aspects of the design here, perhaps apart from brief mentioning for the sake of readability of this PR description.

Some of the aspects mentioned above may be omitted for simple and small changes.

<hr>

This PR has:
- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/incubator-druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

Check the items by putting "x" in the brackets for the done things. Not all of these items apply to every PR. Remove the
items which are not done or not relevant to the PR. None of the items from the checklist above are strictly necessary,
but it would be very helpful if you at least self-review the PR.

<hr>

For reviewers: the key changed/added classes in this PR are `MyFoo`, `OurBar`, and `TheirBaz`.

(Add this section in big PRs to ease navigation in them for reviewers.)

### CODE REVIEW

File: indexing-service/src/main/java/org/apache/druid/indexing/worker/config/WorkerConfig.java [hunk 1]
```diff
@@ -29,6 +29,7 @@
 
 /**
  */
+@SuppressWarnings("SS_SHOULD_BE_STATIC")
```
Comment (clintropolis) [inline indexing-service/src/main/java/org/apache/druid/indexing/worker/config/WorkerConfig.java]: This doesn't seem to be working, travis is still failing with the same error https://travis-ci.org/apache/incubator-druid/jobs/562722959#L1985:
```
[INFO] Total bugs: 3
[ERROR] Unread field: org.apache.druid.indexing.worker.config.WorkerConfig.intermediaryPartitionCleanupPeriodSec; should this field be static? [org.apache.druid.indexing.worker.config.WorkerConfig] At WorkerConfig.java:[line 50] SS_SHOULD_BE_STATIC
[ERROR] Unread field: org.apache.druid.indexing.worker.config.WorkerConfig.intermediaryPartitionDiscoveryPeriodSec; should this field be static? [org.apache.druid.indexing.worker.config.WorkerConfig] At WorkerConfig.java:[line 47] SS_SHOULD_BE_STATIC
[ERROR] Unread field: org.apache.druid.indexing.worker.config.WorkerConfig.version; should this field be static? [org.apache.druid.indexing.worker.config.WorkerConfig] At WorkerConfig.java:[line 39] SS_SHOULD_BE_STATIC
```

Comment (asdf2014) [reply indexing-service/src/main/java/org/apache/druid/indexing/worker/config/WorkerConfig.java]: Please use `@SuppressFBWarnings` instead.

Comment (leventov) [inline indexing-service/src/main/java/org/apache/druid/indexing/worker/config/WorkerConfig.java]: In general, please

1) Make suppressions as local as possible: can suppress for individual fields?
2) Explain a suppression with a comment: mention what does the rule mean ("SS_SHOULD_BE_STATIC" is not self-explanatory) and explain why it should be suppressed in each particular case.

Comment (Fokko) [reply indexing-service/src/main/java/org/apache/druid/indexing/worker/config/WorkerConfig.java]: My mistake, had only a couple of minutes to squeeze out the PR ;)

File: indexing-service/src/main/java/org/apache/druid/indexing/worker/config/WorkerConfig.java [hunk 2]
```diff
@@ -33,30 +34,32 @@
...
 
   @JsonProperty
-  private final long intermediaryPartitionCleanupPeriodSec = 300L;
+  private long intermediaryPartitionCleanupPeriodSec = 300L;
 
   @JsonProperty
-  private final Period intermediaryPartitionTimeout = new Period("P1D");
+  private Period intermediaryPartitionTimeout = new Period("P1D");
 
+  @Nonnull
```
Comment (leventov) [inline indexing-service/src/main/java/org/apache/druid/indexing/worker/config/WorkerConfig.java]: Is this necessary? In general, we add `@EverythingIsNonnullByDefault` on the package declaration in `package-info.java` instead and assume that everything that is not explicitly annotated `@Nullable` is non-null. (Actually, it's violated in a lot of places, and it' a many years-long quest to eliminate those discrepancies, but still.)

`javax.validation.constraints.NotNull` on json properties is something different.

Comment (Fokko) [reply indexing-service/src/main/java/org/apache/druid/indexing/worker/config/WorkerConfig.java]: Thanks for the response. 

The `@Nonnull` it isn't necessary of course. Right now I don't see any `package-info.java` in the indexing part, as `find indexing-service/src/ | grep -i package-info.java` doesn't give anything back.

I've removed the annotation for now. I'll create different PR for adding the `@EverythingIsNonnullByDefault` on the indexing part. Still a lot to be done, but the spotbugs is closing in on the voilations. I'm trying to get rid of all the voilations in an interative way by solving the issues one by one, for example in https://github.com/apache/incubator-druid/pull/8129/

**Comment by leventov** [comment]
@Fokko thanks.

Please, never force-push your PR branch.

**Comment by Fokko** [comment]
I'm force-pushing because of the rebasing onto master. I can also pull master into the branch.

**Comment by leventov** [comment]
You need to merge master into your branch, not rebase. I usually do it as follows:
```
$ git fetch upstream master
$ git merge upstream/master
```
And then resolve conflicts.

**Comment by Fokko** [comment]
I'm fine with merging as well. I usually do a pull instead `git pull github/master`, where github is my upstream.

Most of the projects require a rebase instead of a merge because of history. Mostly because it keeps a linear history, more info here: http://www.bitsnbites.eu/a-tidy-linear-git-history/

**Comment by leventov** [comment]
In Druid, history is linear because of squash-and-merge as the merge action.

**Comment by Fokko** [comment]
That makes sense, I'll merge/pull in the future 👍

**Pull request merged**